VK_App/VK_App/LoginViewController.swift

```diff
@@ -0,0 +1,121 @@
+    @IBOutlet private var passwordTextField: UITextField!
+
+    // MARK: - Life Cycles
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        setupUI()
+    }
+
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        NotificationCenter.default.addObserver(
+            self,
+            selector: #selector(keyboardWasShown),
+            name: UIResponder.keyboardWillShowNotification,
+            object: nil
+        )
+
+        NotificationCenter.default.addObserver(
+            self,
+            selector: #selector(keyboardWillBeHidden),
+            name: UIResponder.keyboardWillHideNotification,
+            object: nil
+        )
+    }
+
+    override func viewWillDisappear(_ animated: Bool) {
+        NotificationCenter.default.removeObserver(
+            self,
+            name: UIResponder.keyboardWillShowNotification,
+            object: nil
+        )
+        NotificationCenter.default.removeObserver(
+            self,
+            name: UIResponder.keyboardWillHideNotification,
+            object: nil
+        )
+    }
+
+    // MARK: - Public Methods
+
+    override func shouldPerformSegue(withIdentifier identifier: String, sender: Any?) -> Bool {
+        let emailText = emailTextField.text
+        let passwordText = passwordTextField.text
+
+        if emailText == Constants.emailText, passwordText == Constants.passwordText {
+            return true
+        } else {
+            alertAction(
+                title: Constants.alertTitleText,
+                message: Constants.alertMessegeText,
+                actionTitle: Constants.alertActionTitleText
+            )
+            return false
+        }
+    }
+
+    // MARK: - Private Methods
+
+    private func setupUI() {
+        createTapScrollView()
+    }
+
+    private func createTapScrollView() {
+        let hideKeyboardGesture = UITapGestureRecognizer(target: self, action: #selector(hideKeyboardAction))
+        scrollView.addGestureRecognizer(hideKeyboardGesture)
+    }
+
+    @objc private func keyboardWasShown(notification: Notification) {
+        guard
+            let info = notification.userInfo as? NSDictionary,
+            let keyboardSize = (info.value(forKey: UIResponder.keyboardFrameEndUserInfoKey) as? NSValue)?.cgRectValue
+            .size
+        else {
+            return
+        }
+        let contentInsets = UIEdgeInsets(top: 0.0, left: 0.0, bottom: keyboardSize.height, right: 0.0)
+        scrollView.contentInset = contentInsets
+        scrollView.scrollIndicatorInsets = contentInsets
+    }
+
+    @objc private func keyboardWillBeHidden() {
+        let contentInsets = UIEdgeInsets.zero
+        scrollView.contentInset = contentInsets
+    }
+
+    @objc private func hideKeyboardAction() {
+        scrollView.endEditing(true)
+    }
+}
+
+// MARK: - Extension UIViewController, добавили метод для создания алерта
+
+extension UIViewController {
```

Comment:
```suggestion
extension UIViewController {
```
это же не для этого контролера значит надо вынести в отдельный файл